Antidote/CallsManager.m

```diff
@@ -0,0 +1,401 @@
+
+@end
+
+@implementation CallsManager
+
+#pragma mark - Lifecycle
+- (instancetype)init
+{
+    self = [super self];
+
+    if (! self) {
+        return nil;
+    }
+
+    _allCallsController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall
+                                                               delegate:self];
+
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"status == %d", OCTCallStatusActive];
+    _allActiveCallsController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall
+                                                                    predicate:predicate
+                                                                     delegate:self];
+
+    predicate = [NSPredicate predicateWithFormat:@"status == %d", OCTCallStatusPaused];
+    _allPausedCallsController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall
+                                                                    predicate:predicate
+                                                                     delegate:self];
+
+    _manager = [AppContext sharedContext].profileManager.toxManager.calls;
+
+    _callNavigation = [UINavigationController new];
+    _callNavigation.view.backgroundColor = [UIColor clearColor];
+    _callNavigation.navigationBarHidden = YES;
+    _callNavigation.modalInPopover = YES;
+    _callNavigation.modalPresentationStyle = UIModalPresentationOverCurrentContext;
+
+    [[AppContext sharedContext].tabBarController presentViewController:_callNavigation animated:YES completion:nil];
+
+    return self;
+}
+
+- (void)dealloc
+{
+    [self.callNavigation dismissViewControllerAnimated:YES completion:nil];
+}
+
+#pragma mark - Public
+
+- (void)callToChat:(OCTChat *)chat
+{
+    if (self.currentCall) {
+        NSAssert(NO, @"We should not be able to make another call yet");
+        return;
+    }
+
+    OCTCall *call = [self.manager callToChat:chat enableAudio:YES enableVideo:NO error:nil];
+
+    if (! call) {
+        [[AppContext sharedContext] killCallsManager];
+    }
+
+    [self dismissAndSwitchViewControllerForCall:call];
+}
+
+- (void)handleIncomingCall:(OCTCall *)call
+{
+    if (self.currentCall && self.pendingIncomingCall) {
+        // We only support showing one incoming call at a time. Reject all others
+        [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:call error:nil];
+    }
+
+    if (self.currentCall) {
+        [self notifyOfIncomingCall:call];
+        return;
+    }
+
+    [self dismissAndSwitchViewControllerForCall:call];
+}
+
+#pragma mark - RBQFetchedResultsControllerDelegate
+
+- (void) controller:(RBQFetchedResultsController *)controller
+    didChangeObject:(RBQSafeRealmObject *)anObject
+        atIndexPath:(NSIndexPath *)indexPath
+      forChangeType:(NSFetchedResultsChangeType)type
+       newIndexPath:(NSIndexPath *)newIndexPath;
+{
+    if (type == NSFetchedResultsChangeUpdate) {
```

Comment:
It would be nicer to change those `if`'s to `switch`

```
switch (type) {
   case NSFetchedResultsChangeUpdate: {
        // code
        break;
    }
    case NSFetchedResultsChangeDelete: {
...
```